doozer/doozerlib/metadata.py

```diff
@@ -401,11 +402,121 @@ async def get_latest_build(self, **kwargs):
+                self.logger.warning('No pinned NVR found for RPM %s and target %s', self.rpm_name, el_target)
+                return None
+
+        else:
+            # The image metadata (or, more likely, the current assembly) has the image
+            # pinned. Return only the pinned NVR. When a child image is being rebased,
+            # it uses get_latest_build to find the parent NVR to use (if it is not
+            # included in the "-i" doozer argument). We need it to find the pinned NVR
+            # to place in its Dockerfile.
+            # Pinning also informs gen-payload when attempting to assemble a release.
+            is_nvr = is_config.nvr
+            if not is_nvr:
+                raise ValueError(f'Did not find nvr field in pinned Image component {self.distgit_key}')
+
+        # strict means raise an exception if not found.
+        return await self.runtime.konflux_db.get_build_record_by_nvr(is_nvr, strict=True)
+
+    async def get_latest_konflux_build(self, assembly: Optional[str] = None,
+                                       outcome: KonfluxBuildOutcome = KonfluxBuildOutcome.SUCCESS,
+                                       el_target: Optional[int] = None,
+                                       honor_is: bool = True,
+                                       completed_before: Optional[datetime] = None,
+                                       extra_patterns: dict = {},
+                                       **kwargs) -> Optional[KonfluxBuildRecord]:
+        """
+        :param assembly: A non-default assembly name to search relative to. If not specified, runtime.assembly
+                         will be used. If runtime.assembly is also None, the search will return true latest.
+                         If the assembly parameter is set to '', this search will also return true latest.
+        :param outcome: one in KonfluxBuildOutcome[FAILURE, SUCCESS, PENDING]
+        :param el_target: specify 'el7', 'el8', etc.
+                          For an RPM, leave as None if you want the true latest.
+                          For an image, leaving as none will default to the RHEL version in the branch
+                          associated with the metadata.
+        :param honor_is: If True, and an assembly component specifies 'is', that nvr will be returned.
+        :param completed_before: cut off timestamp for builds completion time
+        :param extra_patterns: e.g. {'release': 'b45ea65'} will result in adding "AND release LIKE '%b45ea65%'" to the query
+        """
+
+        assert self.runtime.konflux_db is not None, 'Konflux DB must be initialized with GCP credentials'
+        self.runtime.konflux_db.bind(KonfluxBuildRecord)
+
+        # Is the component pinned in config?
+        if honor_is and self.config['is']:
+            if outcome != KonfluxBuildOutcome.SUCCESS:
+                # If this component is defined by 'is', history failures, etc, do not matter.
+                return None
```

Comment:
Here with a non-SUCCESS outcome (including None) and pinned by "is", this method always returns None. This seems to have inconsistent behaviors comparing to normal search.

Reply:
I've tried to follow the same pattern we already had [here](https://github.com/openshift-eng/art-tools/blob/main/doozer/doozerlib/metadata.py#L540-L543), do you see differences in behavior?